### PR TITLE
NIFI-3836 Added check for root user in test that depends on unreadabl…

### PR DESCRIPTION
**Pull request by alopresto**
…e file.

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by alopresto** [comment]
This can be verified by running the Maven test suite in `nifi-commons/nifi-data-provenance-utils` as the `root` user. The process `id -u` should return `0` for the `root` user and a non-`0` value for other users.

**Comment by jfrazee** [comment]
@alopresto I was expecting to see System.getProperty("user.name") used instead of the exec. But I'm a little naive about what the drawback could be; maybe because you could override it and then user.name isn't actually the user.

File: nifi-commons/nifi-data-provenance-utils/src/test/groovy/org/apache/nifi/provenance/CryptoUtilsTest.groovy [hunk 1]
```diff
@@ -88,6 +88,11 @@ class CryptoUtilsTest {
         Cipher.getMaxAllowedKeyLength("AES") > 128
     }
 
+    private static boolean isRootUser() {
+        Process p = Runtime.getRuntime().exec("id -u")
+        p.exitValue() == 0
```
Comment (jfrazee) [inline nifi-commons/nifi-data-provenance-utils/src/test/groovy/org/apache/nifi/provenance/CryptoUtilsTest.groovy]: Isn't this just going to be true whenever `id -u` succeeds?

Comment (alopresto) [inline nifi-commons/nifi-data-provenance-utils/src/test/groovy/org/apache/nifi/provenance/CryptoUtilsTest.groovy]: Yep, excellent catch. I'll fix it to evaluate the response, not the exit code. Thanks.

**Comment by jfrazee** [comment]
Reviewing...

**Comment by jfrazee** [comment]
@alopresto Welp, TestGetFile has similar problems. This one LGTM so question is whether we want to merge this one or expand the scope; thoughts?

**Comment by jfrazee** [comment]
@alopresto +1 will open another JIRA for covering the rest of the problems.

**Pull request closed without merging**